ShortSharing.Tests/ThingServiceTests.cs

```diff
@@ -0,0 +1,136 @@
+using Xunit;
+using Shouldly;
+using AutoMapper;
+using NSubstitute;
+using NSubstitute.ReturnsExtensions;
+using ShortSharing.BLL.Abstractions;
+using ShortSharing.BLL.Mappers;
+using ShortSharing.BLL.Services;
+using ShortSharing.DAL.Abstractions;
+using ShortSharing.DAL.Entities;
+using ShortSharing.BLL.Models;
+
+namespace ShortSharing.Tests;
+
+public class ThingServiceTests
+{
+    private readonly IThingsService _thingsService;
+    private readonly IGenericRepository<ThingEntity> _thingsRepository;
+    private readonly IMapper _mapper;
+
+    public ThingServiceTests()
+    {
+        _thingsRepository = Substitute.For<IGenericRepository<ThingEntity>>();
+
+        _mapper = new MapperConfiguration(cfg =>
+        {
+            cfg.AddProfile(new MapperBllProfile());
+        }).CreateMapper();
+
+        _thingsService = new ThingsService(_thingsRepository, _mapper);
+    }
+
+    [Fact]
+    public async Task GetByIdAsync_InvalidId_ReturnNull()
+    {
+        // Arrange
+        var id = Guid.NewGuid();
```

Comment:
If you use AutoMock already why don't you pass guid id through the method arguments and use [Theory] and [AutoMock] attributes for that? Or you even can pass nay parameter while trying to setup GetByIdAsync method in repository, like this

```
_thingsRepository.GetByIdAsync(Arg.Any<Guid>(), default).ReturnsNull();
```
Please fix that issue for the whole test file.